include/seqan3/io/detail/output_iterator_conversion_adaptor.hpp

```diff
@@ -0,0 +1,217 @@
+ *
+ * ```cpp
+ * using insert_iter = ranges::back_insert_iterator<std::vector<dna4>>;
+ * using out_iter    = seqan3::detail::output_iterator_conversion_adaptor<insert_iter, dna4>;
+ *
+ * std::vector<dna4> out_vec;
+ * out_iter it{insert_iter{out_vec}};
+ *
+ * *it = 'A';
+ * *it = 'C';
+ * *it = 'G';
+ * *it = 'T';
+ *
+ * for (auto && c : out_vec)
+ *     std::cout << to_char(c);
+ * std::cout << '\n'; // prints ACGT
+ * ```
+ */
+template <typename oiter_type, typename alpha_type = char>
+//!\cond
+    requires alphabet_concept<alpha_type> && output_iterator_concept<oiter_type, alpha_type> &&
+        requires (alpha_type t)
+        {
+            { assign_char(t, 'a') }            -> alpha_type;
+            { assign_char(std::move(t), 'a') } -> alpha_type &&;
+        }
+//!\endcond
+class output_iterator_conversion_adaptor
+{
+private:
+    //!\brief The adapted output iterator.
+    oiter_type oiter;
+
+public:
+
+    /*!\name Member types
+     * \{
+     */
+    using value_type        = void;
+    using reference         = void;
+    using pointer           = void;
+    using difference_type   = std::ptrdiff_t;
+    using iterator_category = std::output_iterator_tag;
+    //!\}
+
+    /*!\name Constructor, destructor, assignment
+     * \{
+     */
+    output_iterator_conversion_adaptor() = default;
+    output_iterator_conversion_adaptor(output_iterator_conversion_adaptor const &) = default;
+    output_iterator_conversion_adaptor(output_iterator_conversion_adaptor &&) = default;
+    //!\brief Construction from the adopted iterator.
+    output_iterator_conversion_adaptor(oiter_type _oiter) : oiter{_oiter}
+    {}
+    output_iterator_conversion_adaptor & operator= (output_iterator_conversion_adaptor const &) = default;
+    output_iterator_conversion_adaptor & operator= (output_iterator_conversion_adaptor &&) = default;
+    ~output_iterator_conversion_adaptor() = default;
+    //!\}
+
+    /*!\name Member functions
+     * \{
+     */
+    //!\brief Inserts an object into the associated writable object.
+    constexpr output_iterator_conversion_adaptor & operator= (alpha_type const c)
+    {
+        *oiter = c;
+        ++oiter;
+        return *this;
+    }
+
+    //!\brief Inserts an object into the associated writable object from char.
+    constexpr output_iterator_conversion_adaptor & operator= (char const c)
+    //!\cond
+        requires !same_concept<alpha_type, char>
+    //!\endcond
+    {  // Explicit conversion through char conversion.
+        *oiter = assign_char(alpha_type{}, c);
+        ++oiter;
+        return *this;
+    }
+
+    //!\brief no-op
+    constexpr output_iterator_conversion_adaptor & operator* ()
+    {
+        return *this;
+    }
+
+    //!\brief no-op
+    constexpr output_iterator_conversion_adaptor & operator++ ()
+    {
+        return *this;
+    }
+
+    //!\brief no-op
+    constexpr output_iterator_conversion_adaptor & operator++ (int)
+    {
+        return *this;
+    }
```

Comment:
I think all the member functions should be noexcept. Also please be a little more verbose for the no-ops
`\brief This operator performs no function in output iterators.`